src/js/controllers/imageCtrl-test.js

```diff
@@ -0,0 +1,125 @@
+describe('imagesCtrl', function() {
+    var $rootScope, $controller, imagesCtrl, images, vm, $q;
+
+    beforeEach(module('flickrPOC'));
+
+    beforeEach(module(function($provide){
+        $provide.service('images', function($q){
+            this.loadPics = sinon.stub().returns($q.resolve({message: 'loadPics Working fine'}));
+        });
+
+    }));
+
+    beforeEach(inject(function (_$rootScope_, _$controller_, _images_, _$q_) {
+        $rootScope = _$rootScope_.$new();
+        $controller = _$controller_;
+        images = _images_;
+        $q = _$q_;
+
+        vm = $controller('imagesCtrl');
+
+    }));
+
+    describe('imagesCtrl', function(){
+        beforeEach(function () {
+            images.loadPics.returns($q.resolve(
+                {
+                    contentType: "text/json",
+                    status: "success",
+                    photos: {
+                        photo: [
+                            {
+                            farm: 6,
+                            id: "23780914905",
+                            isfamily: 0,
+                            isfriend: 0,
+                            ispublic: 1,
+                            owner: "138698049@N03",
+                            secret: "0eb2a77a55",
+                            server: "5764",
+                            title: ""
+                        },
+                            {
+                            farm: 1,
+                            id: "23698470321",
+                            isfamily: 0,
+                            isfriend: 0,
+                            ispublic: 1,
+                            owner: "138698049@N03",
+                            secret: "1d4dc255d0",
+                            server: "693",
+                            title: ""
+                        }
+                        ]
+                    }
+                }
+            ));
+        });
+
+        beforeEach(
+            function(){
+                vm.getImages();
+                $rootScope.$apply();
+            }
+        );
+
+        describe('returned flickr images object from http call', function(){
+
+            it('should exist', function () {
+                expect(vm.getImages).to.exist();
```

Comment:
This is a good test, however, I worry that we're actually calling the method (in the `beforeEach()`), before we actually check if it exists. I'd suggest just moving this test to its own `describe()` above so it is not impacted by the `vm.getImages()` call.